rosbridge_library/src/rosbridge_library/protocol.py

```diff
@@ -302,10 +302,8 @@ def serialize(self, msg, cid=None):
                 return bson.BSON.encode(msg)
             else:
                 return json.dumps(msg)
-        except Exception:
-            if cid is not None:
-                # Only bother sending the log message if there's an id
-                self.log("error", "Unable to serialize %s message to client" % msg["op"], cid)
+        except Exception as e:
+            self.log("error", f"Unable to serialize message '{msg}': {e}")
```

Comment:
Without this, I think we can't notice the errors. This time it took a lot of time to find the cause. :pleading_face:

Reply:
interesting - why would `cid` be None?

Reply:
Mm, sorry I'm not sure about that. :cry: